src/box/lua/load_cfg.lua

```diff
@@ -1149,11 +1149,22 @@ local function get_option_from_env(option)
 
     -- This code lean on the existing set of template_cfg
     -- types for simplicity.
-    if param_type:find('table') and raw_value:find(',') then
+    if param_type:find('table') and (raw_value:startswith('{') or
+                                     raw_value:startswith('[')) then
+        return json.decode(raw_value)
+    elseif param_type:find('table') and (raw_value:find(',') or
+                                         raw_value:find('=')) then
         assert(not param_type:find('boolean'))
         local res = {}
         for i, v in ipairs(raw_value:split(',')) do
-            res[i] = tonumber(v) or v
+            local eq = v:find('=')
+            if (eq == nil) then
+                res[i] = tonumber(v) or v
+            else
+                local lhs = string.sub(v, 1, eq - 1)
+                local rhs = string.sub(v, eq + 1)
+                res[lhs] = tonumber(rhs) or rhs
```

Comment:
`TT_XXX=bar=baz,foo` would be accepted as `{bar = 'baz', [2] = 'foo'}`, but I think it shouldn't.

So we can split `raw_value:find('=')` and `raw_value:find(',')` cases.

```lua
elseif param_type:find('table') and raw_value:find('=') then
    <...>
    for _, v in ipairs(raw_value:split(',')) do
        local eq = v:find('=')
        if eq == nil then
            error('XXX')
        end
        <...>
    end
    <...>
elseif param_type:find('table') and raw_value:find(',') then
    <...>
```

Reply:
Done:

```diff
+            local eq = v:find('=')
+            if eq == nil then
+                error(err_msg_fmt:format(env_var_name, option,
+                                         'in `key=value` or `value` format'))
+            end
+            local lhs = string.sub(v, 1, eq - 1)
+            local rhs = string.sub(v, eq + 1)
+            res[lhs] = tonumber(rhs) or rhs
```